src/proto/h1/role.rs

```diff
@@ -635,6 +645,40 @@ impl<'a> Iterator for HeadersAsBytesIter<'a> {
     }
 }
 
+// Write header names as title case. The header name is assumed to be ASCII,
+// therefore it is trivial to convert an ASCII character from lowercase to
+// uppercase. It is as simple as XORing the lowercase character byte with
+// space.
```

Comment:
I believe this will get a couple characters changed completely. Specifically, `|` and `~` will become `\\` and `^`, respectively. Probably not super common, but it'd be a semantic change...

Also, it'd be nice to include some tests of `title_case`, perhaps at the bottom of the file.

Reply:
Oh, I forgot to think of those. A check to see if the byte falls between 97 (`a`) and 172 (`z`) should suffice.

Reply:
I read the octal field by accident from the man page. I fixed it to a `>= b'a'` and `<= b'z'`.